internal/datacoord/compaction_task_clustering.go

```diff
@@ -527,87 +550,107 @@ func (t *clusteringCompactionTask) resetSegmentCompacting() {
 }
 
 func (t *clusteringCompactionTask) processFailedOrTimeout() error {
-	log := log.Ctx(context.TODO())
-	log.Info("clean task", zap.Int64("triggerID", t.GetTaskProto().GetTriggerID()), zap.Int64("planID", t.GetTaskProto().GetPlanID()), zap.String("state", t.GetTaskProto().GetState().String()))
+	return nil
+}
+
+func (t *clusteringCompactionTask) doClean(ctx context.Context) error {
+	log := log.Ctx(ctx).With(zap.Int64("planID", t.GetTaskProto().GetPlanID()))
+	log.Info("clean task", zap.Int64("triggerID", t.GetTaskProto().GetTriggerID()),
+		zap.String("state", t.GetTaskProto().GetState().String()))
 
 	if err := t.sessions.DropCompactionPlan(t.GetTaskProto().GetNodeID(), &datapb.DropCompactionPlanRequest{
```

Comment:
It is imperative to ensure that the DataNode receiver processes this Drop operation idempotently. Additionally, it is essential to capture the merr.ErrNodeNotFound error and consider the DropCompactionPlan operation complete, as the Node has already been decommissioned, rendering the Drop unnecessary.

Reply:
DropCompactionPlan is idempotent on the datanode. The datanode merely keeps the task recorded in memory, so the operation simply attempts to remove the task from memory.

The DropCompactionPlan encapsulated in the session manager already handles the nodeNotFound error, so there is no need to handle it again when calling it externally.